prov/shm/src/smr_util.c

```diff
@@ -393,9 +393,6 @@ int smr_map_to_region(const struct fi_provider *prov, struct smr_map *map,
 	const char *name = smr_no_prefix(peer_buf->peer.name);
 	char tmp[SMR_PATH_MAX];
 
-	if (peer_buf->region)
-		return FI_SUCCESS;
-
```

Comment:
How often do we return FI_SUCCESS from here?  May not be a good idea to remove this check if we hit this all the time.

Reply:
This return should get hit pretty rarely and only on start up. The only case I can think of is if 2 EPs are sharing the same AV and they both get an incoming connection request from the same peer and the threads both try to simultaneously add the peer. It's probably needed but just got moved below the ep name list check to isolate the locking. Ideally it would be above to eliminate the need to go through the EP list but in reality the EP list won't be very long and it would be very rare to get this and it's only on start up so I think it's a reasonable trade off.